spring-batch-neo4j/src/test/java/org/springframework/batch/extensions/neo4j/builder/Neo4jItemWriterBuilderTests.java

```diff
@@ -83,7 +82,7 @@ public void testBasicDelete() {
         IdDescription idDescription = mock(IdDescription.class);
         ExecutableQuery executableQuery = mock(ExecutableQuery.class);
         when(identifierAccessor.getRequiredIdentifier()).thenReturn("someId");
-        when(idDescription.asIdExpression(anyString())).thenReturn(Functions.id(Cypher.anyNode()));
+        when(idDescription.asIdExpression(anyString())).thenReturn(Cypher.anyNode().elementId());
```

Comment:
Here, I had to find an alternative to `Functions.id` because it was no longer accessible with the Neo4J version that Spring Boot is now pulling, but I have to admit I didn't go after all the details.

Is this related to what you just mentioned, @meistermeier? Anything you would do differently?

Reply:
Yes, it is from a product perspective very different. The `elementId` is a complete new (internal) field introduced in Neo4j 5 as a replacement for the `id`.
From a dependency perspective the update might have pulled in new major version of the Cypher DSL (sorry that I have missed this) where we tried to unify the entry point to `Cypher....` for operations and functions.
The only difference, if users don't specify the right Neo4j dialect for the CypherDSL (`Neo4j_5`) is that the `elementId` function will create a fallback to `toString(id(node))` (no index) instead of `elementId(node)` (index support) in the resulting Cypher statement.
It's ok with me because a) we should really switch to elementId and b) it is documented in the CypherDSL. (https://neo4j.github.io/cypher-dsl/2024.2.0/#dialect-support)

Reply:
Thanks for the explanation, @meistermeier! I wait then for your formal approval before merging these changes 🙂